contracts/Marketplace.sol

```diff
@@ -67,11 +71,13 @@ contract Marketplace is Collateral, Proofs {
     _expectProofs(slotId, request.ask.proofProbability, request.ask.duration);
     _submitProof(slotId, proof);
 
-    RequestState storage state = requestState[requestId];
     slot.host = msg.sender;
-    state.slotsFilled += 1;
+    slot.requestId = requestId;
+    context.slotsFilled += 1;
     emit SlotFilled(requestId, slotIndex, slotId);
-    if (state.slotsFilled == request.ask.slots) {
+    if (context.slotsFilled == request.ask.slots) {
+      context.state = RequestState.Started;
+      _extendLockExpiry(requestId, block.timestamp + request.ask.duration);
```

Comment:
When a client submits a storage request, they ask for files to be stored for a `duration`. That means that proofs must be provided until the end of the contract. If we only require each host to provide proofs for a total of `duration` time starting from the time they fill the slot, then unless they filled the last slot (and thus started the contract), they will not be providing proofs until the end of the contract.

I must admit, providing regular proofs for filled slots BEFORE a contract is started seems odd to me. Perhaps removing this requirement would be a potential solution? This means that hosts will only be required to provide regular proofs once the contract has started (in addition to the proof required and submitted to fill the slot).

Reply:
What I mean is that the end of the contract should be `time of request submission + duration`. This means that there are no surprises about the end time of the contract. It is already know when the request is submitted, and does not change based on when the slots were filled.

> What was the reasoning behind providing regular proofs for filled slots BEFORE a contract is started?

Hosts make costs from the moment that they fill a slot, and they're compensated for that because they get paid for every second that they fill a slot. These payments come with an obligation to provide occasional proofs. I saw no reason to treat the period before expiry and the period after expiry different; for a host it means the same effort.